Gems/ROS2/Code/Source/RobotImporter/xacro/XacroUtils.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "RobotImporter/URDF/UrdfParser.h"
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/string/string.h>
+
+namespace ROS2::Utils::xacro
+{
+
+    static const char* kXacroExecutable = "xacro";
+    using Params = AZStd::unordered_map<AZStd::string, AZStd::string>;
+    //! Structure that keeps all artifacts of xacro execution
+    struct ExecutionOutcome
+    {
+        //! parsed URDF from successful xacro's output
+        urdf::ModelInterfaceSharedPtr m_urdfHandle;
+        //! return code of 'xacro' program
+        int m_returnCode{ 255 };
+        //! called program name
+        AZStd::string m_called;
+        //! standard output of xacro execution
+        AZStd::string m_logStandardOutput;
+        //! standard error output of xacro process
+        AZStd::string m_logErrorOutput;
+
+        //! Gets if execution was a success
+        operator bool()
```

Comment:
suggestion: I believe it's preferred to use `explicit` when using operator `bool` to help guard against any weird implicit conversions (this article has a not rundown - https://www.kdab.com/explicit-operator-bool/)

Reply:
It is definitely perfected to use explicit here and something we should add to the [C++ best practices guide](https://github.com/o3de/sig-core/blob/main/governance/C%2B%2B-Best-Practices-Guide.md).

Also this function should be marked **const** as well.
```suggestion
        explicit operator bool() const
```

Reply:
Could you make a PR for this update to the C++ best practices/coding standards @lumberyard-employee-dm? 🙂